docs/guides/tutorials/rest_apis_with_fastapi.rst

```diff
@@ -0,0 +1,535 @@
+
+Prerequisites
+=============
+
+Before we start, make sure you've :ref:`installed <ref_admin_install>` EdgeDB and EdgeDB-CLI. In this tutorial, we'll use Python 3.10 and take advantage of the asynchronous I/O paradigm to communicate with the database more efficiently. A working version of this tutorial can be found `here <https://github.com/edgedb/edgedb-examples/tree/main/fastapi-crud>`_ on GitHub.
+
+
+Install the dependencies
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+To follow along, clone the repository and head over to the ``fastapi-crud`` directory. Create a Python 3.10 virtual environment, activate it, and install the dependencies with this command:
+
+.. code-block:: bash
+    pip install edgedb fastapi httpx[cli] uvicorn
+
+
+Initialize the database
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Now, let's create a new EdgeDB instance for this project. From the root directory, run:
+
+.. code-block:: bash
+    edgedb project init
+
+You should see the following prompts on your console:
+
+::
+    Initializing project...
+
+    Specify the name of EdgeDB instance to use with this project
+    [default: fastapi_crud]:
+    > fastapi_crud
+
+    Do you want to start instance automatically on login? [y/n]
+    > n
+    Checking EdgeDB versions...
+
+Once you've answered the prompts, a new EdgeDB instance called ``fastapi_crud`` will be created. Start the server with the following command:
+
+.. code-block:: bash
+    edgedb instance start fastapi_crud
+
+Connect to the database
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Let's test that we can connect to the newly started instance. To do so, run:
+
+.. code-block:: bash
+    edgedb instance start fastapi_crud
+
+You should be connected to the database instance and able to see a prompt similar to this:
+
+::
+    EdgeDB 1.2+5aecabc (repl 1.1.1+5bb8bad)
+    Type \help for help, \quit to quit.
+    edgedb>
+
+You can start writing queries here. However, we haven't talked about the shape of our data yet.
+
+Schema design
+=============
+
+The event management system will have two entities—**events** and **users**. Each *event* can have an optional link to a *user*. The goal is to create API endpoints that'll allow us to fetch, create, update, and delete the entities while maintaining their relationships.
+
+EdgeDB allows us to declaratively define the structure of the entities. If you've worked with SQLAlchemy or Django ORM, you might refer to these declarative schema definitions as *models*. By default, EdgeDB CLI looks for these schema definitions in the `dbschema/default.esdl` module. You can also create new modules in this directory and call the types from one module to another. However, for now, we'll define our entities in the `default.esdl` module. This is how our datatypes look:
+
+.. code-block:: esdl
+    # dbschema/default.esdl
+
+    module default {
+    abstract type AuditLog {
+      annotation description := "Add 'create_at' and 'update_at' properties to all types.";
```

Comment:
```suggestion
      annotation description := 
          "A mixin type that adds 'create_at' and \
          'update_at' properties to all types.";
```

Reply:
Yeah. I removed the annotations in the tutorial but kept them around in the example application. Curious readers will find their way.

Reply:
AFAIK `updated_at` requires Trigger, which isn't supported as of now.